### ISSUE TITLE
[Tests] Periodically run Firecracker integration tests under sanitizers

### ISSUE THREAD

**Issue by raduiliescu** [firecracker-microvm/firecracker#1662]

**Comment by acatangiu** [comment]
This is now unblocked after merging #2599

**Comment by roypat** [comment]
I'm marking this as parked, as generally, being able to detect memory leaks and undefined behavior in our integration tests would be nice to have, albeit significantly less relevant due to Rust's strong memory safety guarantees. Furthermore, Valgrind does not seem to play nicely with Rust code, as rustc's LLVM backend on x86_64 uses a lot of instructions which valgrind does not support (for example, the SIP hasher used by default in Rust's `HashMap` and `HashSet` uses hardware acceleration not supported by valgrind).

**Comment by JackThomson2** [comment]
This has now been merged in the PR: https://github.com/firecracker-microvm/firecracker/pull/5722
